catalyst_voices/apps/voices/lib/widgets/document_builder/single_dropdown_selection_widget.dart

```diff
@@ -0,0 +1,89 @@
+import 'package:catalyst_voices/widgets/dropdown/voices_dropdown.dart';
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:flutter/material.dart';
+
+class SingleDropdownSelectionWidget extends StatefulWidget {
+  final String value;
+  final List<String> items;
+  final DropDownSingleSelectDefinition definition;
+  final DocumentNodeId nodeId;
+  final String title;
+  final bool isEditMode;
+  final bool isRequired;
+  final ValueChanged<DocumentChange> onChanged;
+
+  const SingleDropdownSelectionWidget({
+    super.key,
+    required this.value,
+    required this.items,
+    required this.definition,
+    required this.nodeId,
+    required this.title,
+    required this.isEditMode,
+    required this.isRequired,
+    required this.onChanged,
+  });
+
+  @override
+  State<SingleDropdownSelectionWidget> createState() =>
+      _SingleDropdownSelectionWidgetState();
+}
+
+class _SingleDropdownSelectionWidgetState
+    extends State<SingleDropdownSelectionWidget> {
+  late List<DropdownMenuEntry<String>> _dropdownMenuEntries;
+  final TextEditingController _textEditingController = TextEditingController();
+  String? value;
+
+  @override
+  void initState() {
+    super.initState();
+    _dropdownMenuEntries = widget.items
+        .map(
+          (e) => DropdownMenuEntry(
+            value: e,
+            label: e,
+          ),
+        )
+        .toList();
+  }
+
+  @override
+  void didUpdateWidget(covariant SingleDropdownSelectionWidget oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.isEditMode != widget.isEditMode &&
+        widget.isEditMode == false) {
+      final value = widget.value;
+      _textEditingController.text = value;
+    }
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _textEditingController.dispose();
```

Comment:
`super.dispose()` goes always last:

```suggestion
    _textEditingController.dispose();
    super.dispose();

```